specification.md

```diff
@@ -2474,6 +2532,218 @@ authenticity of the downloaded file, for example can be the
 The exact signing and verification method is Agent specific. See
 [Code Signing](#code-signing) for recommendations.
 
+### Custom Messages
+
+Status: [Development]
+
+#### Motivation
+
+The OpAMP protocol is intended to cover all aspects of remote management and configuration
+of Agents. If more requirements within this core functionality are identified, it is
+expected that the protocol will be extended to support this additional behavior in future
+releases. However, there are some use cases that require communication with Agents that
+will not by supported by the OpAMP protocol because they are not within the scope of Agent
+remote management or are highly specific to one agent and not generalizable to other agent
+types. CustomCapabilities and CustomMessage allow custom behavior to be implemented
+between a Server and Agent without having to open another connection or define an entirely
+new protocol.
+
+While interoperability between many Agents and Agent Management Servers is a goal of
+OpAMP, additional custom functionality may only be supported by specific Agents and
+Servers. Where appropriate, implementers are encouraged to document their use of
+CustomMessage so that support can be added to additional platforms. If appropriate, custom
+capabilities that are widely supported by Server and Agent implementations may be adopted
+as standard OpAMP capabilities in the future.
+
+#### CustomCapabilities
+
+The Agent and Server should both use this message to signal that they support specific
+custom capabilities. It is supported on ServerToAgent and AgentToServer. This message
+should only be sent when the list of supported custom capabilities changes. When this
+message is sent, the list of supported custom capabilities should be updated to match this
+list. If this message is never sent, it is assumed that no custom capabilities are
+supported.
+
+Capabilities must be identified by a reverse FQDN with optional version information. This
+avoids collisions between capabilities introduced by different organizations. The
+capability name is composed of the reverse FQDN of the organization that created the
+custom capability, a period, and a short name identifying the capability. The version is
+optional and is separated from the capability name by a forward slash, begins with the
+letter "v", and is followed by a number identifying the version.
```

Comment:
Do we need the "v" part and do we need to say that the version is a number? Since we don't interpret it in any way perhaps we simply say the forward slash is follow by version number, e.g.

```suggestion
optional and is separated from the capability name by a forward slash.
```